### ISSUE TITLE
Fix flex item with intrinsic aspect ratio

### ISSUE THREAD

**Issue by shinglyu** [servo/servo#15149]
Quote @stshine 

>flex item with intrinsic aspect ratio
>Item like image that have its intrinsic ratio needs special handling to keep its ratio after flex lengths resolving. Currently servo handle them in https://github.com/servo/servo/blob/master/components/layout/fragment.rs#L488 , which get the size info directly from style, thus won't fit into flexbox. That method needs to be changed, maybe isolating logic for intrinsic aspect ratio is good idea.

Maybe related? https://github.com/servo/servo/pull/14490/files

**Comment by shinglyu** [comment]
https://drafts.csswg.org/css-flexbox/#algo-main-item (?)

**Comment by stshine** [comment]
Yeah, now you can know whether a fragment has intrinsic aspect ratio by the `has_intrinsic_ratio()` method. And you may need to isolate some code from `calculate_replaced_sizes()` and reuse the logic.

**Comment by shinglyu** [comment]
The css-test test cases are blocked by #15402, I'll take a look at that first.
